low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -461,3 +468,53 @@ def get_nw_status(self, nw_interface, interface):
             nw_cable_conn_status = "UNKNOWN"
             logger.exception(self.log.svc_log(err))
         return nw_status, nw_cable_conn_status
+
+    def get_disks_info(self):
+        """Update and return server drive information in specific format."""
+        units_factor_GB = 1000000000
+        overall_usage = {
+                "totalSpace": f'{int(psutil.disk_usage("/")[0])//int(units_factor_GB)} GB',
+                "usedSpace": f'{int(psutil.disk_usage("/")[1])//int(units_factor_GB)} GB',
+                "freeSpace": f'{int(psutil.disk_usage("/")[2])//int(units_factor_GB)} GB',
+                "diskUsedPercentage": psutil.disk_usage("/")[3],
+            }
+        disks = []
+        for disk in Disks().get_disks():
+            uid = disk.path if disk.path else disk.id
+            disk_health = self.get_health_template(uid, True)
+            health_data = disk.get_health()
+            health = "OK" if health_data['SMART_health'] else "Fault"
+            self.set_health_data(disk_health, health, specifics=[{"SMART": health_data}])
+            disks.append(disk_health)
+        disks_health_info = [{
+            "overall_usage": overall_usage,
+            "disk_count": len(disks),
+            "disks": disks,
+            "last_updated": int(time.time())
+        }]
+        return disks_health_info
+
+    def get_psu_info(self):
+        """Update and return PSU information in specific format."""
+        psus = []
+        psu_mapping = {
+            "PS1 Status": "PSU1",
+            "PS2 Status": "PSU2"
+        }
+        uids = {}
+        for psu in self._ipmi.get_sensor_list_by_type("Power Supply"):
+            psu_name, sensor, *_ = psu
+            uids[psu_mapping[psu_name]] = f"#0x{sensor.strip('h').lower()}"
+        response, _, _ = SimpleProcess("dmidecode -t 39").run()
+        matches = re.finditer(r"System Power Supply\n(\s*.*\n)"
+                              r"\s*Location: (?P<location>.*)\n"
+                              r"(\s*.*\n){7}\s*Status: (?P<status>.*)\n"
+                              r"(\s*.*\n){2}\s*Plugged: (?P<plugged>.*)",
+                              response.decode())
+        for match in matches:
+            psu_data = match.groupdict()
+            data = self.get_health_template(f'Power Supply {uids[psu_data["location"]]}', True)
+            health = "OK" if (psu_data["status"] == "Present, OK") else "Fault"
+            self.set_health_data(data, health, specifics=psu_data)
+            psus.append(data)
```

Comment:
add debug log to record the generated data as doing for other modules

Reply:
Done